lib/active_model_serializers/key_transform.rb

```diff
@@ -66,5 +66,26 @@ def underscore(value)
     def unaltered(value)
       value
     end
+
+    # Transforms string to selected case
+    # Accepts string in any case: 'camel', 'undescore', 'dashed'.
+    #
+    # @example:
+    #   transform('SomeClass', :underscore) => 'some_class'
+    #   transform('some_class', :snake) => 'someClass'
+    #   etc...
+    def transform(string, type)
+      string = string.underscore
+      case type.to_sym
+      when :dashed
+        string.dasherize
+      when :camel
+        string.camelize
+      when :snake
+        string.camelize(:lower)
+      else
+        string
+      end
+    end
```

Comment:
It pre-existed.. And fixing merge conflicts didn't fix feature dup. Am hoping someone to finish:)

B mobile phone

> On Jun 9, 2016, at 9:16 PM, Ben Mills <notifications@github.com> wrote:
> 
> In lib/active_model_serializers/key_transform.rb:
> 
> > +    #   transform('SomeClass', :underscore) => 'some_class'
> > +    #   transform('some_class', :snake) => 'someClass'
> > +    #   etc...
> > +    def transform(string, type)
> > +      string = string.underscore
> > +      case type.to_sym
> > +      when :dashed
> > +        string.dasherize
> > +      when :camel
> > +        string.camelize
> > +      when :snake
> > +        string.camelize(:lower)
> > +      else
> > +        string
> > +      end
> > +    end
> Is there a reason this doesn't leverage the existing transforms?
> Why does there need to be a jsonapi_type_transform and a key_transform? Are there situations you'd really want them to differ?
> Can type.map! ... from lib/active_model_serializers/adapter/json_api/resource_identifier.rb just use KeyTransform.send?
> Isn't "snake (underscore)" snake_case, "camel (lower)" camelCase, and camel (pascal) CamelCase?
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub, or mute the thread.
>